addons/payment_payulatam/controllers/main.py

```diff
@@ -3,17 +3,61 @@
 import logging
 import pprint
 
+import werkzeug.exceptions
+
 from odoo import http
+from odoo.exceptions import ValidationError
 from odoo.http import request
 
 _logger = logging.getLogger(__name__)
 
 
 class PayuLatamController(http.Controller):
     _return_url = '/payment/payulatam/return'
+    _webhook_url = '/payment/payulatam/webhook'
 
     @http.route(_return_url, type='http', auth='public', methods=['GET'])
     def payulatam_return(self, **data):
         _logger.info("handling redirection from PayU Latam with data:\n%s", pprint.pformat(data))
+        self._verify_signature(data)
         request.env['payment.transaction'].sudo()._handle_feedback_data('payulatam', data)
         return request.redirect('/payment/status')
+
+    @http.route(_webhook_url, type='http', auth='public', methods=['POST'], csrf=False)
+    def payulatam_webhook(self, **data):
+        _logger.info("handling confirmation from PayU Latam with data:\n%s", pprint.pformat(data))
+        state_pol = data.get('state_pol')
+        if state_pol == '4':
+            lapTransactionState = 'APPROVED'
+        elif state_pol == '6':
+            lapTransactionState = 'DECLINED'
+        elif state_pol == '5':
+            lapTransactionState = 'EXPIRED'
```

Comment:
Wouldn't it be better to keep 
```py
else:
    lapTransactionState = 'ERROR'
```
then ?

Reply:
I don't think so. Because if you assign 'ERROR' to the `lapTransactionState`, you are speaking about the transaction state, and not an error in the notification, and should we ever manage this case, it'll put your transaction in an error state.
Example: let's imagine PayU messed something up with its notification, there is no `lapTransactionState`, you put the transaction in error state, and immediately after, PayU send back another notification, this time including the code that could be `APPROVED`. Your transaction state won't go from `error` to `done`. And there is the problem.
Again, since we don't manage the `ERROR` state, this won't happen anyway. We just want to be clear.

Reply:
A good compromise could be
```py
else:
    lapTransactionState = f'INVALID state_pol {state_pol}'
```
as it allows using the error handling code of `payment.transaction._process_feedback_data` and still propagate the information on `state_pol` rather than only `lapTransactionState`.